### ISSUE TITLE
Foreign modules aren't recognized in bloop

### ISSUE THREAD

**Issue by nrktkt** [com-lihaoyi/mill#876]
Creating a build like 

```scala
// ./build.sc
import mill._, scalalib._
import $file.bar.build
object foo extends ScalaModule {
    def scalaVersion = "2.13.1"
    def moduleDeps = Seq(bar.build.bar, baz)
}
object baz extends ScalaModule {
    def scalaVersion = "2.13.1"
}

// ./bar/build.sc
import mill._, scalalib._
object bar extends ScalaModule {
    def scalaVersion = "2.13.1"
}
```

Causes warnings in bloop 
> Missing projects 'bar', may cause compilation issues in project 'foo'

and indeed, metals has issues with highlighting although compiling from mill directly works OK.

Tested on linux with mill 0.6.3

**Comment by tgodzik** [comment]
How hard would it be to create an additional target for the imported module? I think it's possible to fix it in Metals, but that would be complicated and I don't foresee any contributors having time to work on it soon.

**Comment by ajrnz** [comment]
#1291 should fix this problem. It wasn't picking up foreign modules up because it wasn't looking transitively in `moduleDeps`

**Comment by lefou** [comment]
Fixed in #1291 for `moduleDeps`

More thorough fix in #1802 which also inspects `compileModuleDeps`
